var/spack/repos/builtin/packages/gettext/package.py

```diff
@@ -84,8 +84,10 @@ def configure_args(self):
             config_args.append('--disable-curses')
 
         if '+libxml2' in spec:
-            config_args.append('--with-libxml2-prefix={0}'.format(
+            config_args.append('CPPFLAGS=-I{0}/include'.format(
                 spec['libxml2'].prefix))
+            config_args.append('LDFLAGS=-L{0} -Wl,-rpath,{0}'.format(
+                spec['libxml2'].libs.directories[0]))
```

Comment:
Looking back at this, it worries me that it wasn't working before. Spack's compiler wrappers add these exactly flags. Can you check the build log and see if it actually uses Spack's compiler wrappers?

Reply:
I just did a side-by-side comparison: with and without the fix.  It worked with the fix, and not without.  I'm including build logs here.

[gettext-badd.txt](https://github.com/spack/spack/files/2024230/gettext-badd.txt)
[gettext-good.txt](https://github.com/spack/spack/files/2024231/gettext-good.txt)

Reply:
It looks like it's not using Spack's compiler wrappers.  I have no idea why.  Note that on this system, I'm using a hand-built GCC, which requires some modules to properly function.

```
compilers:
- compiler:
    modules: [other/comp/gcc-5.3-sp3]#,linux-suse_linux11-x86_64/binutils-2.27-gcc-5.3.0-p7ewifc]
    operating_system: suse_linux11
    paths:
      cc: /usr/local/other/SLES11.3/gcc/5.3.0/bin/gcc
      cxx: /usr/local/other/SLES11.3/gcc/5.3.0/bin/g++
      f77: /usr/local/other/SLES11.3/gcc/5.3.0/bin/gfortran
      fc: /usr/local/other/SLES11.3/gcc/5.3.0/bin/gfortran
    spec: gcc@5.3.0
```
I also have in my `.bashrc`:
```
# Load Basics for Spack, Compiler and MPI
module load other/comp/gcc-5.3-sp3
module load other/mpi/openmpi/1.10.1-gcc-5.3-sp3
module load binutils-2.29.1-gcc-5.3.0-dimunje
module load curl-7.59.0-gcc-5.3.0-qjbgogv
module load git-2.17.0-gcc-5.3.0-air7gv6
```

Reply:
This looks like the culprit:
```
checking if /usr/local/other/SLES11.3/gcc/5.3.0/bin/gcc supports -fno-rtti -fno-exceptions... no
```
If it were using Spack's compiler wrappers, it would look like:
```
checking whether /Users/Adam/spack/lib/spack/env/clang/clang accepts -g... yes
```
Any idea why it would be ignoring Spack's compiler wrappers for you?